base_layer/wallet_ffi/src/lib.rs

```diff
@@ -2786,7 +2789,15 @@ pub unsafe extern "C" fn wallet_get_completed_transactions(
         .block_on((*wallet).transaction_service.get_completed_transactions());
     match completed_transactions {
         Ok(completed_transactions) => {
-            for (_id, tx) in &completed_transactions {
+            // The frontend specification calls for completed transactions that have not yet been mined to be
+            // classified as Pending Transactions. In order to support this logic without impacting the practical
+            // definitions and storage of a MimbleWimble CompletedTransaction we will remove CompletedTransactions with
+            // the Completed and Broadcast states from the list returned by this FFI function
+            for tx in completed_transactions
+                .values()
+                .filter(|ct| ct.status != TransactionStatus::Completed)
+                .filter(|ct| ct.status != TransactionStatus::Broadcast)
```

Comment:
Two passes through the list, when you could combine the predicate into one expression...

Reply:
Is this two passes? Isn't it chained 1 iterated item at a time?